resolwe/flow/models/data.py

```diff
@@ -395,10 +401,10 @@ def create_entity(self):
                 entity = None
 
             if entity:
-                entity.data.add(self)
-                # Inherit collections from entity.
-                for collection in entity.collections.all():
-                    collection.data.add(self)
+                self.entity = entity
+                # Inherit collection from entity.
+                self.collection = entity.collection
```

Comment:
It should probably be the other way around. Collection is given when data is created. Creating data may trigger entity creation, so entity should inherit a collection from data (in `Entity.objects.create`). Shouldn't it?